graph/src/main/scala/net/spals/dsa/graph/Graph.scala

```diff
@@ -50,6 +50,16 @@ case class WeightedGraph[T, W](vertices: Set[Vertex[T]], edgeFunc: (Vertex[T], V
     edgeFunc.apply(v1, v2).isDefined
   }
 
+  def removeEdge(v1: Vertex[T], v2: Vertex[T]): WeightedGraph[T, W] = {
```

Comment:
We could do cyclic weighted graphs?
Or make non weight graphs which is a weighted graph of 1 naturally.